src/cloudai/workloads/nemo_run/cloudai_nemorun.py

```diff
@@ -199,13 +642,194 @@ def cloudai_recipe() -> run.Partial:
+            num_nodes=1,
+            accelerator="gpu",
+            max_steps=10,
+            limit_test_batches=50,
+            limit_val_batches=32,
+            log_every_n_steps=10,
+            use_distributed_sampler=False,
+            plugins=run.Config(
+                nl.MegatronMixedPrecision,
+                autocast_enabled=False,
+                grad_reduce_in_fp32=False,
+                params_dtype=torch.bfloat16,
+                pipeline_dtype=torch.bfloat16,
+                precision="bf16-mixed",
+            ),
+            strategy=run.Config(
+                nl.MegatronStrategy,
+                tensor_model_parallel_size=8,
+                pipeline_model_parallel_size=8,
+                context_parallel_size=2,
+                virtual_pipeline_model_parallel_size=12,
+                sequence_parallel=True,
+                pipeline_dtype=torch.bfloat16,
+                gradient_as_bucket_view=True,
+                ckpt_async_save=True,
+                ckpt_include_optimizer=True,
+                ckpt_parallel_load=True,
+                ddp=run.Config(
+                    DistributedDataParallelConfig,
+                    check_for_nan_in_grad=True,
+                    grad_reduce_in_fp32=True,
+                    overlap_grad_reduce=True,
+                    overlap_param_gather=True,
+                    average_in_collective=True,
+                ),
+            ),
+            num_sanity_val_steps=0,
+            val_check_interval=500,
+            max_epochs=10,
+            callbacks=[
+                run.Config(
+                    MegatronCommOverlapCallback,
+                    tp_comm_overlap=True,
+                    overlap_grad_reduce=True,
+                    overlap_param_gather=True,
+                ),
+                timing_callback(),
+            ],
+        ),
+        optim=run.Config(
+            nl.MegatronOptimizerModule,
+            config=run.Config(
+                OptimizerConfig,
+                lr=1e-4,
+                bf16=True,
+                params_dtype=torch.bfloat16,
+                use_distributed_optimizer=True,
+                weight_decay=0,
+            ),
+        ),
+        resume=run.Config(
+            nl.AutoResume,
+            resume_if_exists=True,
+            resume_ignore_no_checkpoint=True,
+            resume_past_end=True,
+        ),
+    )
     return recipe
 
 
 if __name__ == "__main__":
     mode = os.getenv("CLOUDAI_NEMO_TASK")
-    print(f"Running in mode {mode}")
+
+    supported_recipes = {
+        "llama3_8b": cloudai_llama3_8b_recipe,
+        "llama3_70b": cloudai_llama3_70b_recipe,
+        "llama3_405b": cloudai_llama3_405b_recipe,
+        "nemotron3_8b": cloudai_nemotron3_8b_recipe,
+        "nemotron4_15b": cloudai_nemotron4_15b_recipe,
+        "nemotron4_340b": cloudai_nemotron4_340b_recipe,
+    }
+
+    recipe_name = os.getenv("CLOUDAI_NEMO_RECIPE")
+
+    if recipe_name not in supported_recipes:
```

Comment:
This is a pretty hardcore limitation IMO. I would like if we could still run default recipes as well. Imagine a config with just a recipe name defined, so everything else is set as default, why wouldn't that work?

Reply:
I'm not sure we need to duplicate the warning here, it is hard to track changes in two places to be the same. But it is up to you.